lib/reporters/GruntCoverage.js

```diff
@@ -0,0 +1,64 @@
+define([
+    'dojo/node!istanbul/lib/collector',
+    'dojo/node!istanbul/lib/object-utils',
+    'dojo/node!grunt'
+], function (Collector, utils, grunt) {
+    function GruntCoverage(config) {
+        this.config = config || {};
+        config.threshold = config.threshold || { overall: 75 };
+        this._collector = new Collector();
+    }
+
+    function enforceThreshold(threshold, actual, expected) {
+        var name = threshold[0].toUpperCase() + threshold.slice(1),
+            success = actual >= expected,
+            placing = success ? 'above' : 'below',
+            passed = success ? 'PASSED! ' : 'FAILED! ',
+            status = success ? 'ok' : 'error',
+            msg = passed + name + ': ' + actual + '%, ' + placing + ' threshold of ' + expected + '%';
+
+        grunt.log[status](msg);
+        return success;
+    }
+
+    GruntCoverage.prototype.coverage = function (sessionId, coverage) {
+        this._collector.add(coverage);
+    };
+
+    GruntCoverage.prototype.runEnd = function () {
+        var collector = this._collector,
+            summaries = [],
+            total = 0,
+            overall = 0,
+            success = true,
+            finalSummary;
+
+        collector.files().forEach(function (file) {
+            summaries.push(utils.summarizeFileCoverage(collector.fileCoverageFor(file)));
+        });
+
+        finalSummary = utils.mergeSummaryObjects.apply(null, summaries);
+
+        for (var key in finalSummary) {
+            if (finalSummary[key].pct) {
+                if (this.config.threshold.hasOwnProperty(key)) {
+                    success = success && enforceThreshold(key, finalSummary[key].pct, this.config.threshold[key]);
+                }
+                overall += finalSummary[key].pct;
```

Comment:
I don’t think you can just combine function, statement, line, and branch coverage and then divide to get an average of like this since they are measures of different incompatible units.